### ISSUE TITLE
booking-up-for-beauty: add link and improve task with AnniversaryDate()

### ISSUE THREAD

**Issue by Jasstkn** [exercism/go#2243]
Hi.

I have two questions/proposals:
1. Do you think it will be useful to add link to https://gobyexample.com/time-formatting-parsing as well? It was helpful for me to solve this task.
2. I think we can make it a bit harder for passing AnniversaryDate. For example, it can take a year as a parameter:
- if this argument is empty, it should print the current year anniversary, 
- if it's not - print required year anniversary. 

In my opinion it can be useful to do more calculations in terms of working with time resources. 

Let me know what do you think :)

**Comment by andrerfcsantos** [comment]
> Do you think it will be useful to add link to https://gobyexample.com/time-formatting-parsing as well? It was helpful for me to solve this task.

Parsing time strings in Go can be confusing as it works differently from other languages, so more help is always welcome!

> if this argument is empty, it should print the current year anniversary

Assuming the year parameter would be an int, what exactly do you mean by "empty"?

The task currently makes the student get the current year which is something new to the task and makes the student create a date from scratch using all parameters., which is also new in the task. If we were to add on this, I think we should make the student make something else with the date functions.

Alternatively, we can make a new task, especially if there's something the exercise currently doesn't talk about that you think it should.

**Comment by Jasstkn** [comment]
> > Do you think it will be useful to add link to https://gobyexample.com/time-formatting-parsing as well? It was helpful for me to solve this task.
> 
> Parsing time strings in Go can be confusing as it works differently from other languages, so more help is always welcome! And Go By Example is always a great source to include.
> 
> > if this argument is empty, it should print the current year anniversary
> 
> Assuming the year parameter would be an int, what exactly do you mean by "empty"?
> 
> The task currently makes the student get the current year which is something new to the task and makes the student create a date from scratch using all parameters., which is also new in the task. If we were to add on this, I think we should make the student make something else with the date functions.
> 
> Alternatively, we can make a new task, especially if there's something the exercise currently doesn't talk about that you think it should.

Thanks for reply. I'll create a PR with new link then. 

Regarding improvement, I was thinking about something like that:
```
func AnniversaryDateV2(year ...int) time.Time {
	if len(year) == 0 {
		return ...
	}
	return ...
}
```

But it seems to be not ideal at all so maybe you're right and it's better to have a separate task in scope of working with time..
We can get inspired by example here: https://www.sohamkamani.com/golang/time/#duration---adding-and-subtracting-time, for e.g. we have scheduled appointment at the salon and we want to notify our client how much time has left before the appointment time.

**Comment by junedev** [comment]
@andrerfcsantos Can this issue be closed or is there something else open to be addressed?

**Comment by andrerfcsantos** [comment]
@junedev While the first point for the link was addressed by #2246, there's still the question if we should add something to the exercise.

**Comment by Jasstkn** [comment]
> @junedev While the first point for the link was addressed by #2246, there's still the question if we should add something to the exercise.

Let me know if you have any thoughts about what I wrote here: https://github.com/exercism/go/issues/2243#issuecomment-1152218842

**Comment by Jasstkn** [comment]
@junedev Okay, I don't have any objections.
